File: src/118PascalsTriangle/118PascalsTriangle.go
package main

func generate(numRows int) [][]int {
	if numRows <= 0 {
		return nil
	}
	var results [][]int
	result := make([]int, numRows)
	result[0] = 1
	for i := 0; i < numRows; i++ {
		for j := i; j >= 0; j-- {
			if j-1 >= 0 {
				result[j] += result[j-1]
			}
		}
		r := make([]int, i+1)
		copy(r, result[:i+1])
		results = append(results, r)
	}
	return results
}

func main() {

}

File: src/118PascalsTriangle/118PascalsTriangle.rs
impl Solution {
    pub fn generate(num_rows: i32) -> Vec<Vec<i32>> {
        let mut results = vec![];
        for i in 0..num_rows{
            if i==0 {
                results.push(vec![1]);
            }else{
                let mut result = vec![];
                for j in 0..i+1 {
                    let l = if j-1<0 {0} else {results[(i-1) as usize][(j-1) as usize]};
                    let r = if j>i-1 {0} else {results[(i-1) as usize][j as usize]};
                    result.push(l+r);
                }
                results.push(result);
            } 
        }
        results
    }
}
